src/docker/job_control.py

```diff
@@ -1,132 +1,245 @@
-bucket=os.getenv("OUTPUT_BUCKET")
-bucket='apbs-test-output'
-queue=os.getenv("JOB_QUEUE_NAME")
-queue='apbs-test-job-q'
-
-
-def get_items(sqs,qurl):
-  loop=0
-
-  items = sqs.receive_message( QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=qtimeout)
-  
-  while 'Messages' not in items:
-    loop+=1
-    if loop==max_tries:
-      return 0
-    print ("Waiting ....")
-    time.sleep(retry_time)
-    items = sqs.receive_message( QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=qtimeout)
-  
-  return items
-
-def update_state(s3,jobid,jobtype,status,output):
-  ups3=boto3.resource('s3')
-  objectfile=jobid+'/'+jobtype+'-status.json'
-  s3obj = ups3.Object(bucket, objectfile)
-  statobj:dict = json.loads(s3obj.get()['Body'].read().decode('utf-8')) 
-  
-  statobj[jobtype]['status']=status
-  statobj[jobtype]['endTime']=time.time()
-  ## FIX
-  statobj[jobtype]['outputFiles']=output
-  
-  object_response:dict = s3.put_object(
-                              Body=json.dumps(statobj),
-                              Bucket=bucket,
-                              Key=objectfile
-  )
-
-def run_code(job,s3):
-  out=1
-  try:
-    job_info:dict = json.loads(job) 
-    if 'job_id' not in job_info:
-      return 1
-  except:
-    print ("Not a json q item")
+"""Software to run apbs and pdb2pqr jobs."""
+
+from os import chdir, getenv, listdir, makedirs, system
+from datetime import datetime
+from enum import Enum
+from json import dumps, loads
+from logging import getLogger
+from time import sleep, time
+from shutil import rmtree
+from typing import Any, Dict, List
+from urllib import request
+from boto3 import client, resource
+
+_LOGGER = getLogger(__name__)
+Q_TIMEOUT = 300
+AWS_REGION = "us-west-2"
+MAX_TRIES = 60
+RETRY_TIME = 15
+
+MEM_PATH = "/dev/shm/test/"
+S3_BUCKET = getenv("OUTPUT_BUCKET")
+QUEUE = getenv("JOB_QUEUE_NAME")
+
+
+class JOBTYPE(Enum):
+    """The valid values for a job's type."""
+
+    APBS = 1
+    PDB2PQR = 2
+
+
+class JOBSTATUS(Enum):
+    """The valid values for a job's status."""
+
+    COMPLETE = 1
+    RUNNING = 2
+
+
+def get_messages(sqs: client, qurl: str) -> Any:
+    """Get SQS Messages from the queue.
+
+    :param sqs:  S3 output bucket for the job being updated
+    :type sqs:  boto3.client connection
+    :param qurl:  URL for the SNS Queue to listen for new messages
+    :return:  List of messages from the queue
+    :rtype:  Any
+    """
+    loop = 0
+
+    items = sqs.receive_message(
+        QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=Q_TIMEOUT
+    )
+
+    while "Messages" not in items:
+        loop += 1
+        if loop == MAX_TRIES:
+            return None
+        _LOGGER.info("Waiting ....")
+        sleep(RETRY_TIME)
+        items = sqs.receive_message(
+            QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=Q_TIMEOUT
+        )
+
+    return items
+
+
+def update_status(
+    s3client: client,
+    jobid: str,
+    jobtype: JOBTYPE,
```

Comment:
this is actually a type str being passed in.
which means that line 86 should not have .name

Reply:
I think this is fixed too.